parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetRecordWriter.java

```diff
@@ -112,6 +153,10 @@ public ParquetRecordWriter(
   @Override
   public void close(TaskAttemptContext context) throws IOException, InterruptedException {
     internalWriter.close();
+    // release after the writer closes in case it is used for a last flush
+    if (codecFactory != null) {
```

Comment:
similar comment as below, if close above can throw we should put it in a try and call release from a finally block. This method can still throw the checked exceptions, no real need to re-wrap them.